### PR TITLE
Optimize conv2d_op_internal forward path

### PR DESCRIPTION
**Pull request by beru**
Hi, this PR optimizes `conv2d_op_internal` function.

### CODE REVIEW

File: tiny_dnn/core/kernels/conv2d_op_internal.h [hunk 1]
```diff
@@ -58,54 +58,59 @@ inline void conv2d_op_internal(const tensor_t &in_data,
...
+               serial_size_t idx;
+               idx               = params.weight.get_index(0, 0, id * o + inc);
+               const float_t *pw = &W[idx];
+               idx               = params.in_padded.get_index(0, 0, inc);
+               const float_t *pi = &in[idx];
+               float_t *pa2      = pa;
+               for (serial_size_t y = 0; y < oh; y++) {
+                 const float_t *pi2 = pi;
+                 for (serial_size_t x = 0; x < ow; x++) {
+                   const float_t *pi3 = pi2;
```
Comment (Randl) [inline tiny_dnn/core/kernels/conv2d_op_internal.h]: Is it possible to somehow improve naming? I'm trying to read the code and getting lost very fast.  A big amount of two-letter variables is confusing.

Comment (edgarriba) [inline tiny_dnn/core/kernels/conv2d_op_internal.h]: it comes from historical code, but indeed we can do better here

Comment (beru) [inline tiny_dnn/core/kernels/conv2d_op_internal.h]: @Randl Thanks for the review. I will change names of the local variables later.

**Comment by edgarriba** [comment]
@Randl @bhack @karandesai-96 can I merge?

**Comment by beru** [comment]
@edgarriba Please wait to merge. I'll try to change variable names.

File: tiny_dnn/core/kernels/conv2d_op_internal.h (hunk 1 above)
Comment (beru) [inline tiny_dnn/core/kernels/conv2d_op_internal.h]: @Randl Can you check the code again? If it's still not easily readable, I'll stop using two-letter variables.

**Pull request merged**